modules/renter/download.go

```diff
@@ -519,3 +520,49 @@ func (r *Renter) DownloadHistory() []modules.DownloadInfo {
 	}
 	return downloads
 }
+
+// ClearDownloadHistory clears the renter's download history inclusive
+// of the provided before and after timestamps
+func (r *Renter) ClearDownloadHistory(after, before time.Time) error {
+	if err := r.tg.Add(); err != nil {
+		return err
+	}
+	defer r.tg.Done()
+	r.downloadHistoryMu.Lock()
+	defer r.downloadHistoryMu.Unlock()
+
+	// Check to confirm there are downloads to clear
+	if len(r.downloadHistory) == 0 {
+		return nil
+	}
+
+	// Timestamp validation
+	if before.Before(after) && !before.IsZero() {
+		return errors.New("before timestamp can not be newer then after timestamp")
+	}
+
+	// Clear download history if both before and after timestamps are zero values
+	if before.IsZero() && after.IsZero() {
+		r.downloadHistory = r.downloadHistory[:0]
+		return nil
+	}
+
+	// Find and return downloads that are not within the given range
+	if after.IsZero() {
+		after = r.downloadHistory[0].staticStartTime
+	}
+	if before.IsZero() {
```

Comment:
This is not intuitive, clear everything before time 0 should indicate that you clear nothing at all.

Reply:
If `before` or `after` `IsZero()` that means a value wasn't submitted so this is checking for that condition and handling it accordingly.  I can add more comments to the function to clarify that.  From a user's POV they would just be submitting a value for `before` or `after` if they wanted to use that functionality, they don't need to submit a zero value for the other parameter.

Reply:
what about creating a const of type time called `infinity` and set it equal to `time.Unix(math.MaxInt64,0)`  which is equal to `219250468-12-04 15:30:07 +0000 UTC`

Reply:
This should be sufficient for infinity as long as we aren't around in the year 219250468

Reply:
also `time.Unix(0,math.MaxInt64)` is the year 2262 if we think that is sufficient and then we wouldn't have to worry about overflow errors if someone used `.UnixNano()` on the const

Reply:
- [ ] Create modules.EndOfTime const